ChatCore/Classes/ChatCore.swift

```diff
@@ -514,6 +514,20 @@ extension ChatCore {
     }
 }
 
+// MARK: - Loading messages
+extension ChatCore {
+    open func getMessages(conversation id: EntityIdentifier, request: MessagesRequest, completion: @escaping (Result<[MessageUI], ChatError>) -> Void) {
+        networking.getMessages(conversation: id, request: request) { result in
+            switch result {
+            case let .success(messages):
+                completion(.success(messages.map { $0.uiModel }))
```

Comment:
You should dispatch it to main queue cause the completion from the networking can come on any queue